docs/api/modules/State-Management.md

```diff
@@ -0,0 +1,153 @@
+
+**Contents**
+* [Globals](#globals)
+* [Higher Order Components](#higher-order-components)
+* [Shared Ducks](#shared-ducks)
+
+## Globals
+
+### `global.BROWSER`
+
+**Runs On**
+* ✅ Server
+* ✅ Browser
+
+**Shape**
+
+```js
+global.BROWSER; // Boolean
+```
+
+The `global.BROWSER` is provided in the Server and Browser environments to determine if the code is currently being executed on the Server or the Browser.
+
+## Higher Order Components
+
+[Holocron Modules](../API.md#modules) use [Higher Order Components (HOC)](https://reactjs.org/docs/higher-order-components.html) to add behaviors regarding 1) when a Module loads, 2) connecting a Module with its [Reducer(s)](https://redux.js.org/basics/reducers/) to a Redux Store (similar to [Redux `connect`](https://react-redux.js.org/api/connect)) and 3) adding runtime validations for a Module.
+
+**Contents**
+* [`holocronModule`](#holocronmodule)
+
+### `holocronModule`
+
+Please see [`holocronModule`](https://github.com/americanexpress/holocron/blob/master/packages/holocron/API.md#holocronmodule) in the Holocron API.
+
+## Shared Ducks
+
+One App uses a default set of shared Redux [Ducks](https://github.com/erikras/ducks-modular-redux) (e.g. a design pattern for logical groupings of Reducers, Actions, and Selectors that depend on each other) to manage the state of core features (e.g. Loading Language Packs or Holocron Modules) on the Server and Browser environments.
+
+The following is an overview of the shape One App provides:
+
+```
+{
+  config,
+  errorReporting,
+  modules,
+  error,
+  browser,
+  rendering,
+  redirection,
+  holocron,
+  intl
+}
+```
+
+The following API definitions describe the Ducks responsible for the state shape described above.
+
+**Contents**
+* [`config`](#config-duck)
+* [`errorReporting`](#errorreporting-duck)
+* [`error`](#error-duck)
+* [`browser`](#browser-duck)
+* [`rendering`](#rendering-duck)
+* [`redirection`](#redirection-duck)
+* [`holocron`](#holocron-duck)
+* [`intl`](#intl-duck)
+
+### `config` Duck
+The `config` Duck lists a subset of the environment variables set on the Server.
+
+**Contents:**
+* [State Shape](#state-shape)
+* Action Creators
+  * [`setConfig`](#setconfig)
+
+#### State Shape
+
+```js
+const state = new Map({
+  config: new Map({
+    // URL where the Browser sends client side errors to.
+    reportingUrl: String,
+    // URL where the One App static assets are located.
+    cdnUrl: String,
+    // The exact filename for the locale file used in the Browser.
+    localeFilename: String,
+    // Name of the Holocron Module that serves as the entry point to your application.
+    rootModuleName: String,
+    // ... Settings from provideStateConfig key values will land here.
+    [provideStateConfigSettingName]: String,
```

Comment:
Don't we parse these values, so they are not always strings?

Reply:
They are always strings. The values that land in `config` are set based on environmental variable and browser/server. 

Parsing `stateConfig.js`:
https://github.com/americanexpress/one-app/blob/master/src/server/utils/stateConfig.js#L127-L140

Calling `setConfig` action:
https://github.com/americanexpress/one-app/blob/0c64dcd7b76018868c37101e50ef2228c25a3eec/src/server/middleware/sendHtml.js#L287

Reply:
I was thinking of the env var parsing https://github.com/americanexpress/one-app/blob/master/src/server/utils/stateConfig.js#L23-L37

We should definitely make this more flexible

Reply:
The env vars are parsed, so they are not always strings

https://github.com/americanexpress/one-app/blob/master/src/server/utils/stateConfig.js#L23-L37

The config from the root module should be more flexible